app/services/api/v1x0/catalog/copy_portfolio.rb

```diff
@@ -19,15 +19,19 @@ def process
         private
 
         def make_copy
-          @portfolio.dup.tap do |new_portfolio|
-            new_portfolio.name = @name || ::Catalog::NameAdjust.create_copy_name(@portfolio.name, Portfolio.all.pluck(:name), Portfolio::MAX_NAME_LENGTH)
+          Portfolio.transaction do
+            @portfolio.dup.tap do |new_portfolio|
+              new_portfolio.name = @name || ::Catalog::NameAdjust.create_copy_name(@portfolio.name, Portfolio.all.pluck(:name), Portfolio::MAX_NAME_LENGTH)
 
-            duplicate_icon(@portfolio, new_portfolio) if @portfolio.icon_id.present?
+              duplicate_icon(@portfolio, new_portfolio) if @portfolio.icon_id.present?
 
-            new_portfolio.save!
+              new_portfolio.save!
 
-            copy_portfolio_items(new_portfolio.id)
+              copy_portfolio_items(new_portfolio.id)
+            end
           end
+        rescue => e
```

Comment:
We should not rescue without an exception type, and switching any error to `InvalidSurvey` is likely not correct either.

How does the `rescue` help, instead of just letting the original error percolate up.

Reply:
@gmcculloug you are right. I was just testing 1 weird scenario where I added this line of code: `raise ActiveRecord::Rollback` into the `copy_portfolio_items` method. In this case the copy action was not finished, but the final result looked weird. As you can see bellow, I was redirected into the page where should be a new Portfolio, but it was not copied. When I raised my own exception or 1 of the catalog expections all worked fine. It made me wonder about difference between them.

![Screenshot from 2020-07-09 17-00-04](https://user-images.githubusercontent.com/19405716/87056392-d6a53e00-c205-11ea-8efc-9613a079bdb3.png)